src/smt/optimization_solver.h

```diff
@@ -122,32 +163,49 @@ class OptimizationSolver
    * Constructor
    * @param parent the smt_solver that the user added their assertions to
    **/
-  OptimizationSolver(SmtEngine* parent);
-  ~OptimizationSolver();
+  OptimizationSolver(SmtEngine* parent)
+      : d_parent(parent), d_objectives(), d_results()
+  {
+  }
+  ~OptimizationSolver() = default;
+
+  /** Runs the optimization loop for the pushed objective **/
```

Comment:
Please specify in this comment whether it is the last pushed objective, or that you are assuming that only a single objective was pushed (in that case assert this in the implementation).

Reply:
the behavior currently is to always optimize the first one. 
I don't want to add this to comments because multiobj algorithms are coming right after so that I don't need to change all this comments.

Reply:
If you assume it is single you should assert this and comment on this. No matter if a follow up PR will change this behavior, this is what goes into master with this PR, so this should be properly asserted and documented.